src/test/FreeC/Backend/Agda/Converter/TypeDeclTests.hs

```diff
@@ -54,7 +54,7 @@ testConvertDataDecls =
       shouldConvertTypeDeclsTo
         (Recursive ["data List a = Nil | Cons a (List a)"])
         [ "data List (Shape : Set)(Pos : Shape → Set)(a : Set) : {Size} → Set where"
-        ++ "  nil : ∀ {i} → List Shape Pos a {↑ i}"
+        ++ "  nil : List Shape Pos a"
```

Comment:
Yes, i did some test with the old Ada backend. This new implementation drops the whole annotation if it isn't needed. I tested multiple data types from the `free-proving-agda-code` repository and have not found any problems.
> The old Agda back end annotated all constructor with a Size annotation,
> because it was convenient to do so. The new back end places annotations
> only if needed. This change does not effect the behavior in Agda.